File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairsRecur(int n, vector<int>& arr){
        if (n == 1 || n == 0)
            return 1;
        if (arr[n] != -1)
            return arr[n];
        arr[n]= climbStairsRecur(n-1 , arr) + climbStairsRecur(n-2, arr);
        return arr[n];
    }
    int climbStairs(int n) {
        vector<int>arr(n+1, -1);
        return climbStairsRecur(n, arr);
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        arr= [-1]*(n+1)
        return self.climbStairsRecursive(n, arr)
    
    def climbStairsRecursive(self, n : int, arr: list) -> int:
        if n == 1 or n == 0 : return 1
        if arr[n] != -1:
            return arr[n]
        arr[n] = self.climbStairsRecursive(n-1 , arr) + self.climbStairsRecursive(n-2, arr)
        return arr[n]